pkg/apply/apply.go

```diff
@@ -49,6 +53,30 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 		return errors.Wrapf(err, "could not retrieve existing %s", objDesc)
 	}
 
+	if obj.GetKind() == "Namespace" {
+		ns := &v1.Namespace{}
+		if err := kscheme.Scheme.Convert(existing, ns, nil); err != nil {
+			return errors.Wrapf(err, "could not convert namespace %s", objDesc)
+		}
+		if ns.Status.Phase == v1.NamespaceTerminating {
+			log.Printf("finalize Namespace %s", objDesc)
+			ns.Spec.Finalizers = []v1.FinalizerName{}
+			// Get a config to for client-go
+			cfg, err := config.GetConfig()
+			if err != nil {
+				return errors.Wrapf(err, "could not create rest config")
+			}
+			clientset, err := kubernetes.NewForConfig(cfg)
+			if err != nil {
+				return errors.Wrapf(err, "could not create clientset")
+			}
+			if _, err := clientset.CoreV1().Namespaces().Finalize(ns); err != nil {
```

Comment:
I may be wrong, but I think you can do instead:
if _, err := client.CoreV1().Namespaces().Finalize(ns); err 

And avoid instantiating a new client, removing the lines 64-72

Reply:
The `client` here is not a k8s client-go clientset, but a controller-runtime client which doesn't support the `finalize` call of the namespace object. https://github.com/kubernetes-sigs/controller-runtime/issues/573

Reply:
That's not going to work. He wants to access the `Finalize` method, which is not exposed by `k8sclient.Client`, but there might be a smoother way to do this than instantiating a new clientset like this?